crates/bevy_render/src/base_bundle.rs

```diff
@@ -0,0 +1,59 @@
+use bevy_ecs::prelude::Bundle;
+use bevy_transform::prelude::{GlobalTransform, Transform};
+
+use crate::view::{ComputedVisibility, Visibility};
+
+/// A [`Bundle`] with the following [`Component`](bevy_ecs::component::Component)s:
+/// * [`Visibility`] and [`ComputedVisibility`], which describe the visibility of an entity
+/// * [`Transform`] and [`GlobalTransform`], which describe the position of an entity
+///
+/// * To show or hide an entity, you should set its [`Visibility`].
+/// * To get the computed visibility of an entity, you should get its [`ComputedVisibility`].
+/// * To place or move an entity, you should set its [`Transform`].
+/// * To get the global position of an entity, you should get its [`GlobalTransform`].
+/// * For hierarchies to work correctly, you must have all four components.
+///   * You may use the [`BaseBundle`] to guarantee this.
+#[derive(Bundle, Debug, Default)]
+pub struct BaseBundle {
+    /// The visibility of the entity.
+    pub visibility: Visibility,
+    /// The computed visibility of the entity.
+    pub computed: ComputedVisibility,
+    /// The transform of the entity.
+    pub local: Transform,
+    /// The global transform of the entity.
+    pub global: GlobalTransform,
+}
+
+impl BaseBundle {
+    /// Creates a new [`BaseBundle`] from a [`Transform`].
+    ///
+    /// This initializes [`GlobalTransform`] as identity, and visibility as visible
+    #[inline]
+    pub const fn from_transform(transform: Transform) -> Self {
+        BaseBundle {
+            local: transform,
+            // Note: `..Default::default()` cannot be used here, because it isn't const
+            ..Self::visible_identity()
+        }
+    }
+
+    /// Creates a new identity [`BaseBundle`], with no translation, rotation, and a scale of 1
+    /// on all axes.
+    #[inline]
+    pub const fn visible_identity() -> Self {
```

Comment:
I'd prefer this as an associated const.

Reply:
same, I copied from https://github.com/bevyengine/bevy/blob/8eb0440f1ed0f075ed488e7efa36ba0f8bb37cad/crates/bevy_transform/src/lib.rs#L65